Public/Src/Engine/Processes/SandboxConnectionES.cs

```diff
@@ -0,0 +1,214 @@
+            m_workerThread.Priority = ThreadPriority.Highest;
+            m_workerThread.Start();
+        }
+
+        /// <summary>
+        /// Disposes the sandbox connection and release the resources in the interop layer, when running tests this can be skipped
+        /// </summary>
+        public void Dispose()
+        {
+            if (!IsInTestMode)
+            {
+                ReleaseResources();
+            }
+        }
+
+        /// <summary>
+        /// Releases all resources and cleans up the interop instance too
+        /// </summary>
+        public void ReleaseResources()
+        {
+            Sandbox.DeinitializeEndpointSecuritySandbox(m_esConnectionInfo);
+            m_workerThread.Join();
+        }
+
+        /// <summary>
+        /// Starts listening for reports from the EndpointSecurity sandbox
+        /// </summary>
+        private void StartReceivingAccessReports()
+        {
+            Sandbox.AccessReportCallback callback = (Sandbox.AccessReport report, int code) =>
+            {   
+                if (code != Sandbox.ReportQueueSuccessCode)
+                {
+                    var message = "EndpointSecurity event delivery failed with error: " + code;
+                    throw new BuildXLException(message, ExceptionRootCause.MissingRuntimeDependency);
+                }
+                
+                // Stamp the access report as it has been dequeued at this point
+                report.Statistics.DequeueTime = Sandbox.GetMachAbsoluteTime();
+
+                // Update last received timestamp
+                Volatile.Write(ref m_lastReportReceivedTimestampTicks, DateTime.UtcNow.Ticks);
+
+                // Remember the latest enqueue time
+                Volatile.Write(ref m_reportQueueLastEnqueueTime, report.Statistics.EnqueueTime);
+
+                // The only way it can happen that no process is found for 'report.PipId' is when that pip is
+                // explicitly terminated (e.g., because it timed out or Ctrl-c was pressed)
+                if (m_pipProcesses.TryGetValue(report.PipId, out var process))
+                {
+                    // if the process is found, its ProcessId must match the RootPid of the report.
+                    if (process.ProcessId != report.RootPid)
+                    {
+                        throw new BuildXLException("The process id from the lookup did not match the file access report process id", ExceptionRootCause.FailFast);
+                    }
+                    else
+                    {
+                        process.PostAccessReport(report);
+                    }
+                }
+            };
+
+            Sandbox.ObserverFileAccessReports(ref m_esConnectionInfo, callback, Marshal.SizeOf<Sandbox.AccessReport>());
+        }
+
+        /// <inheritdoc />
+        public bool NotifyUsage(uint cpuUsage, uint availableRamMB)
+        {
+            // TODO: Will we need this?

```

Comment:
Nope.  The kext uses this stuff for dynamic throttling, which in the end was not very useful